ArduCopter/autoyaw.cpp

```diff
@@ -264,3 +270,30 @@ float Mode::AutoYaw::rate_cds() const
     // return zero turn rate (this should never happen)
     return 0.0f;
 }
+
+void Mode::AutoYaw::update_weathervane(const int16_t pilot_yaw, const int16_t roll_cdeg, const int16_t pitch_cdeg)
+{
+#if !HAL_MINIMIZE_FEATURES
+    if (copter.g2.weathervane.should_weathervane(pilot_yaw, roll_cdeg, pitch_cdeg)) {
+        if (mode() != AUTO_YAW_WEATHERVANE) {
+                set_mode(AUTO_YAW_WEATHERVANE);
+            }
+
+            // force weathervane controller to relax on landing, in case min height is not set
+            copter.g2.weathervane.set_relax(copter.ap.land_complete || copter.ap.land_complete_maybe);
+
+            // set yaw rate 
+            _yaw_rate_cds = copter.g2.weathervane.get_yaw_rate_cds(roll_cdeg, pitch_cdeg);
```

Comment:
So the output is a rate instead of an angle?  .. and I guess this explains why we have a separate gain.  With the introduction of SCurves we support a simultaneous target angle and rate.

Reply:
yes, output is rate.  We convert the roll angle to a yaw rate via the gain.